federatedscope/core/regularizer/l2_regularizer.py

```diff
@@ -0,0 +1,36 @@
+from federatedscope.register import register_regularizer
+from torch.nn import Module
+
+import torch
+
+REGULARIZER_NAME = "l2_regularizer"
+
+
+class L2Regularizer(Module):
+    """Returns the l2 norm of weight
+
+        Arguments:
+            p (int): The order of norm.
+            tensor_before: The original matrix or vector
+            tensor_after: The updated matrix or vector
+
+        Returns:
+            Tensor: the norm of the given udpate.
+    """
+    def __init__(self):
+        super(L2Regularizer, self).__init__()
+
+    def forward(self, ctx):
+        l2_norm = 0.
+        for param in ctx.model.parameters():
```

Comment:
An interesting discussion about whether l2 regularization should be applied to beta and gamma of BN layers: https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994.
IMO, we can keep the implementation same as that in `torch.norm`

Reply:
The current `torch.norm` will calculate the l2 norm for all parameters including bn weight and bias. 
For now, we use the parameter `skip_bn` to aovid calculating l2-norm for bn layer(with 'bn' in its name).

Reply:
Conflicts are solved